tokendito/user.py

```diff
@@ -517,6 +522,65 @@ def prompt_role_choices(aut_tiles):
     return selected_role
 
 
+def get_role_expiration(role_response={}):
+    """Get the expiration from the role response.
+
+    :param role_response: Assume Role response dict
+    :return expiration
+    """
+    try:
+        return role_response["Credentials"]["Expiration"]
+    except (KeyError, TypeError) as err:
+        logger.error(f"Could not retrieve expiration: {err}")
+        sys.exit(1)
+
+
+def check_profile_expiration(config):
+    """Check profile expiration and exit if still valid.
+
+    :param config: Config object
+    """
+    if not config.user["use_profile_expiration"]:
+        return
+
+    profile = config.aws["profile"]
+
+    profile_expiration_str = config.okta["profile_expiration"]
+    if not profile_expiration_str:
+        logger.warning(f"Expiration unavailable for config profile {profile}. ")
+        return
+
+    profile_expiration = datetime.fromisoformat(profile_expiration_str)
+    now = datetime.now(timezone.utc)
```

Comment:
Probably not necessary but if it isn't guaranteed to return utc.
```suggestion
    now = datetime.now(profile_expiration.tzinfo if profile_espiration.tzinfo else timezone.utc)
```